### PR TITLE
The slowest PS instance deletes outdated checkpoint directory.

### LINKED ISSUE #1809: "No such file or directory" occasionally happens when the PS save checkpoint.
``` 
File "/elasticdl/python/worker/worker.py", line 579, in report_gradient_to_ps
    res = report_future.result()
  File "/usr/local/lib/python3.6/dist-packages/grpc/_channel.py", line 625, in result
    raise self
grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:
        status = StatusCode.UNKNOWN
        details = "Exception calling application: [Errno 2] No such file or directory: '/testdata/ckpt/20200309-19/version-26000/variables-0-of-3.ckpt'"
        debug_error_string = "{"created":"@1583757719.258002676","description":"Error received from peer ipv4:11.149.168.3:2222","file":"src/core/lib/surface/call.cc","file_line":1056,"grpc_message":"Exception calling application: [Errno 2] No such file or directory: '/testdata/elasticdl/kept/dban_lookup/20200309-19/version-26000/variables-0-of-3.ckpt'","grpc_status":2}"
```

### PR DESCRIPTION
**Pull request by workingloong**
Fix #1809

### CODE REVIEW

File: elasticdl/python/common/save_utils.py [hunk 1]
```diff
@@ -141,33 +142,31 @@ def save(
...
-                # Remove the directory if empty
-                if not os.listdir(checkpoint_version_dir):
-                    os.rmdir(checkpoint_version_dir)
-
-    def get_latest_checkpoint_version(self):
-        """Get the latest checkpointed model version"""
-        if not self._checkpoint_list:
-            raise RuntimeError("No model checkpoint available")
-        return self._checkpoint_list[-1].version
+        while len(self._checkpoint_dir_list) > self._max_versions:
```
Comment (QiJune) [inline elasticdl/python/common/save_utils.py]: It seems that it's no need to use`while` here. Once the sloweset save the latest checkpoint, the oldest checkpoint will be removed

Comment (workingloong) [reply elasticdl/python/common/save_utils.py]: Yeah, `if` is enough.

**Review by terrytangyuan — APPROVED** [review]

File: elasticdl/python/common/save_utils.py [hunk 2]
```diff
@@ -141,33 +142,29 @@ def save(
...
-        if not self._checkpoint_list:
-            raise RuntimeError("No model checkpoint available")
-        return self._checkpoint_list[-1].version
+        if len(self._checkpoint_dir_list) > self._max_versions:
+            old_version_dir = self._checkpoint_dir_list[0]
+
+            # Some PS instances have not saved checkpoint shard files of
+            # the version if invalid. And the slowest PS will remove the
+            # old version checkpoint.
+            if self.check_checkpoint_valid(old_version_dir):
```
Comment (brightcoder01) [inline elasticdl/python/common/save_utils.py:164]: How do you make sure that this is the slowest PS instance?

Comment (workingloong) [reply elasticdl/python/common/save_utils.py:164]: After the slowest PS instance saves its model shard into checkpoint folder for a version, the shards are complete. `check_checkpoint_valid` returns true only when the shards are complete.

**Pull request merged**